### DISCUSSION TITLE
Problem Updating to V7.3

### DISCUSSION THREAD

**Discussion by xocula** [sanni/cartreader#324] [Need help]
Hi, I get this error when I try to update to V7.3 from V6.6:  

redefinition of void printSTR(String st, int x, int y)

What could be the problem?

**Comment by sanni** [comment]
How does your options.h look like? Maybe you got two lines where // is removed.

````
// Remove // in front of the line with your hardware version
//#define HW4
#define HW3
//#define HW2
//#define HW1
//#define SERIAL_MONITOR
````

**Comment by xocula** [comment]
This is how I put it, I am new in this, its my first time with arduino and maybe i am wrong.

/******************************************
   Libraries
 *****************************************/
// Options
#include "options.h"

//******************************************
// CHOOSE HARDWARE VERSION
//******************************************
//#define HW4
#define HW3
//#define HW2
//#define HW1
//#define SERIAL_MONITOR

// Basic Libs
#include <SPI.h>
#include <Wire.h>
#include <avr/pgmspace.h>
#include <avr/wdt.h>

**Comment by xocula** [comment]
Here is the complete error that I get, in case this helps.


Arduino:1.8.16 (Windows 10), Tarjeta:"Arduino Mega or Mega 2560, ATmega2560 (Mega 2560)"

C:\Users\juana\AppData\Local\Temp\arduino_modified_sketch_370055\N64.ino: In function 'void printSTR(String, int, int)':

N64:1434:6: error: redefinition of 'void printSTR(String, int, int)'

 void printSTR(String st, int x, int y)

      ^~~~~~~~

C:\Users\juana\AppData\Local\Temp\arduino_modified_sketch_370055\N64.ino:950:6: note: 'void printSTR(String, int, int)' previously defined here

 void printSTR(String st, int x, int y)

      ^~~~~~~~

C:\Users\juana\AppData\Local\Temp\arduino_modified_sketch_370055\N64.ino: In function 'void controllerTest_OLED()':

N64:1504:53: error: 'WHITE' was not declared in this scope

           display.drawLine(22 + 0, 10, 22 + 84, 10, WHITE);

                                                     ^~~~~

C:\Users\juana\AppData\Local\Temp\arduino_modified_sketch_370055\N64.ino:1504:53: note: suggested alternative: 'WDIE'

           display.drawLine(22 + 0, 10, 22 + 84, 10, WHITE);

                                                     ^~~~~

                                                     WDIE

N64:1546:53: error: 'WHITE' was not declared in this scope

           display.drawLine(22 + 0, 50, 22 + 84, 50, WHITE);

                                                     ^~~~~

C:\Users\juana\AppData\Local\Temp\arduino_modified_sketch_370055\N64.ino:1546:53: note: suggested alternative: 'WDIE'

           display.drawLine(22 + 0, 50, 22 + 84, 50, WHITE);

                                                     ^~~~~

                                                     WDIE

N64:1624:50: error: 'BLACK' was not declared in this scope

           display.drawPixel(22 + prevStickX, 40, BLACK);

                                                  ^~~~~

N64:1626:53: error: 'WHITE' was not declared in this scope

           display.drawLine(22 + 0, 10, 22 + 83, 10, WHITE);

                                                     ^~~~~

C:\Users\juana\AppData\Local\Temp\arduino_modified_sketch_370055\N64.ino:1626:53: note: suggested alternative: 'WDIE'

           display.drawLine(22 + 0, 10, 22 + 83, 10, WHITE);

                                                     ^~~~~

                                                     WDIE

N64:1627:19: error: 'class U8G2_ST7567_OS12864_F_4W_HW_SPI' has no member named 'drawRect'; did you mean 'drawRBox'?

           display.drawRect(22 + 0, 15, 22 + 59, 21, WHITE);

                   ^~~~~~~~

                   drawRBox

N64:1748:104: error: 'WHITE' was not declared in this scope

                 display.drawLine(xax + upx / 4, yax - upy / 4, xax + uprightx / 4, yax - uprighty / 4, WHITE);

                                                                                                        ^~~~~

C:\Users\juana\AppData\Local\Temp\arduino_modified_sketch_370055\N64.ino:1748:104: note: suggested alternative: 'WDIE'

                 display.drawLine(xax + upx / 4, yax - upy / 4, xax + uprightx / 4, yax - uprighty / 4, WHITE);

                                                                                                        ^~~~~

                                                                                                        WDIE

N64:1913:53: error: 'WHITE' was not declared in this scope

             display.drawLine(22 + 0, 9, 22 + 83, 9, WHITE);

                                                     ^~~~~

C:\Users\juana\AppData\Local\Temp\arduino_modified_sketch_370055\N64.ino:1913:53: note: suggested alternative: 'WDIE'

             display.drawLine(22 + 0, 9, 22 + 83, 9, WHITE);

                                                     ^~~~~

                                                     WDIE

C:\Users\juana\AppData\Local\Temp\arduino_modified_sketch_370055\NES.ino: In function 'void setMapper()':

NES:1157:42: error: 'WHITE' was not declared in this scope

         display.drawLine(20, 30, 30, 30, WHITE);

                                          ^~~~~

C:\Users\juana\AppData\Local\Temp\arduino_modified_sketch_370055\NES.ino:1157:42: note: suggested alternative: 'WDIE'

         display.drawLine(20, 30, 30, 30, WHITE);

                                          ^~~~~

                                          WDIE

NES:1158:42: error: 'BLACK' was not declared in this scope

         display.drawLine(40, 30, 50, 30, BLACK);

                                          ^~~~~

NES:1162:42: error: 'BLACK' was not declared in this scope

         display.drawLine(20, 30, 30, 30, BLACK);

                                          ^~~~~

NES:1163:42: error: 'WHITE' was not declared in this scope

         display.drawLine(40, 30, 50, 30, WHITE);

                                          ^~~~~

C:\Users\juana\AppData\Local\Temp\arduino_modified_sketch_370055\NES.ino:1163:42: note: suggested alternative: 'WDIE'

         display.drawLine(40, 30, 50, 30, WHITE);

                                          ^~~~~

                                          WDIE

NES:1167:42: error: 'BLACK' was not declared in this scope

         display.drawLine(20, 30, 30, 30, BLACK);

                                          ^~~~~

NES:1169:42: error: 'WHITE' was not declared in this scope

         display.drawLine(60, 30, 70, 30, WHITE);

                                          ^~~~~

C:\Users\juana\AppData\Local\Temp\arduino_modified_sketch_370055\NES.ino:1169:42: note: suggested alternative: 'WDIE'

         display.drawLine(60, 30, 70, 30, WHITE);

                                          ^~~~~

                                          WDIE

exit status 1

redefinition of 'void printSTR(String, int, int)'

**Comment by sanni** [comment]
Nononono, no, that's not how it works.

You copied it into Cart_Reader.ino but the version definition is already in the options.h tab. You need to edit it there.

![choosehardware](https://user-images.githubusercontent.com/19169494/151654538-9853c974-46ba-44b0-bac3-48eb566e9bf0.jpg)

**Comment by xocula** [comment]
Thank you, I knew I was doing something wrong. Its my first time with arduino and I apreciate quite a lot your help

**Comment by sanni** [comment]
I have updated to guides to include the picture now. 👍
